rules/no-useless-promise-resolve-reject.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+const {methodCallSelector} = require('./selectors/index.js');
+
+const RETURN_RESOLVE = 'return-resolve';
+const RETURN_REJECT = 'return-reject';
+const YIELD_RESOLVE = 'yield-resolve';
+const YIELD_REJECT = 'yield-reject';
+const messages = {
+	[RETURN_RESOLVE]: 'Prefer `return value` over `return Promise.resolve(value)`.',
+	[RETURN_REJECT]: 'Prefer `throw error` over `return Promise.reject(error)`.',
+	[YIELD_RESOLVE]: 'Prefer `yield value` over `yield Promise.resolve(value)`.',
+	[YIELD_REJECT]: 'Prefer `throw value` over `yield Promise.reject(value)`.',
+};
+
+const promiseResolveOrRejectSelector = methodCallSelector({
+	object: 'Promise',
+	methods: ['resolve', 'reject'],
+	argumentsLength: 1,
```

Comment:
On second thought, we should not limit arguments length to `1`, but make sure handle them correctly

```js
return Promise.resolve()
// -> 
return;

yield Promise.resolve()
// -> 
yield;

return Promise.reject()
// -> 
throw undefind; // Can't fix to `throw;` it's invalid, we can throw undefined or not fix at all.
```

Reply:
What about multiple arguments? Should the fix be a suggestion instead so the user can correct their mistake (the extra args could cause side effects despite being unnecessary for Promise.resolve/reject)?

Reply:
yes, multiple arguments, report but no fix.